src/EditorFeatures/Core/Implementation/AutomaticCompletion/BraceCompletionSessionProvider.BraceCompletionSession.cs

```diff
@@ -69,34 +69,18 @@ public void Start()
 
             private void Start(CancellationToken cancellationToken)
             {
-                // this is where the caret should go after the change
                 var pos = TextView.Caret.Position.BufferPosition;
                 var beforeTrackingPoint = pos.Snapshot.CreateTrackingPoint(pos.Position, PointTrackingMode.Negative);
 
                 var snapshot = SubjectBuffer.CurrentSnapshot;
                 var closingSnapshotPoint = ClosingPoint.GetPoint(snapshot);
 
-                if (closingSnapshotPoint.Position < 1)
```

Comment:
The code to create the edit and caret location was moved to IBraceCompletionService.GetBraceCompletion
(implementation is in AbstractBraceCompletionService)